include/pybind11/cast.h

```diff
@@ -434,11 +438,19 @@ struct string_caster {
     cast(const StringType &src, return_value_policy /* policy */, handle /* parent */) {
         const char *buffer = reinterpret_cast<const char *>(src.data());
         auto nbytes = ssize_t(src.size() * sizeof(CharT));
-        handle s = decode_utfN(buffer, nbytes);
-        if (!s) {
-            throw error_already_set();
+        if (std::is_same_v<CharT, uint8_t>
```

Comment:
`is_same_v` is not introduced until C++17 and we don't have this function available in the backports in the common.h yet.